app/admin/views.py

```diff
@@ -49,10 +49,16 @@ def invite_user():
     """Invites a new user to create an account and set their own password."""
     form = InviteUserForm()
     if form.validate_on_submit():
-        user = User(role=form.role.data,
-                    first_name=form.first_name.data,
-                    last_name=form.last_name.data,
-                    email=form.email.data)
+        role_choice = form.role.data.name
+        print role_choice
+        if role_choice == 'Vendor':
+            user = Vendor(email=form.email.data)
+        elif role_choice == 'Merchant':
+            user = Merchant(email=form.email.data)
+        else:
+            user = User(role=form.role.data,
```

Comment:
Slightly confused. These options *are* in the dropdown--we're getting them *from* the dropdown (`form.role.data.name`). Agreed on the `else` block, it should be `elif role_choice == 'Admin'`, but then our case coverage is incomplete (still have that pesky default role). I think we definitely need to just finish this PR, and then branch off again for an entirely new task of "nuking the default user/registration" since that involves multiple steps of (1) auto-registering an administrator (2) removing on-site registration and (3) cleaning up loose ends like these `else` blocks.

If you have an idea for (1) I'll gladly implement it and then do the rest real quick, but should that be on this branch?